admin/testbrew.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Script executed by Buildbot on the Mac mini build slave, this instantiates the
+OS X Virtual Machine to a snapshot with a clean homebrew install, then executes
+via SSH the brew update, install and test commands to verify a successful
+installation of a homebrew recipe.
+
+The host machine must have:
+        * A VMWare OS X VM available at a particular location
+
+The guest virtual machine must have:
+        * A snapshot with a particular name, homebrew-clean
+        * Homebrew installed and available
+"""
+
+from _preamble import TOPLEVEL, BASEPATH
+
+import os
+import sys
+import urllib2
+
+from twisted.python.usage import Options, UsageError
+
+from subprocess import check_output, CalledProcessError
+
+from flocker.provision._install import run_with_fabric, task_test_homebrew
+
+
+YOSEMITE_VMX_PATH = (
+    '"'
+    "{home}/Documents/Virtual Machines.localized/"
+    "OS X 10.10.vmwarevm/OS X 10.10.vmx"
+    '"'
+).format(home=os.getenv("HOME"))
+
+YOSEMITE_SNAPSHOT = "homebrew-clean"
+
+VM_USERNAME = "ClusterHQVM"
+
+VM_HOST = "10.0.126.88"
+
+
+class TestBrewOptions(Options):
+    description = ("Start a OS X Virtual Machine and test a brew recipe "
+                   "installatiion.")
+
+    optParameters = [
+        ['vmhost', 'h', VM_HOST,
+         'IP address or hostname of the Virtual Machine'],
+        ['vmuser', 'u', VM_USERNAME,
+         'Username for the Virtual Machine'],
+        ['vmpath', 'p', YOSEMITE_VMX_PATH,
+         'Full path to the Virtual Machine image'],
+        ['vmsnapshot', 's', YOSEMITE_SNAPSHOT,
+         'Snapshot identifier for the Virtual Machine'],
+    ]
+
+    synopsis = ('Usage: testbrew.py <recipe URL> [--vmhost <host>] '
```

Comment:
Our other `admin` scripts follow a pattern which in this case would be something like:

test-homebrew:

```
#!/usr/bin/env python
# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
"""
Test a Homebrew script.
"""

from _preamble import TOPLEVEL, BASEPATH

import sys

if __name__ == '__main__':
    from admin.test_homebrew import main
    main(sys.argv[1:], top_level=TOPLEVEL, base_path=BASEPATH)
```

test_homebrew.py: What is here now.

It is probably worth glancing over one or two to keep consistency. Note that at least `upload-rpms` is an early script which is different but there is WIP to change that.